src/Interpreters/ReplaceQueryParameterVisitor.cpp

```diff
@@ -69,7 +69,14 @@ void ReplaceQueryParameterVisitor::visitQueryParameter(ASTPtr & ast)
             " because it isn't parsed completely: only {} of {} bytes was parsed: {}",
             value, type_name, ast_param.name, read_buffer.count(), value.size(), value.substr(0, read_buffer.count()));
 
-    ast = addTypeConversionToAST(std::make_shared<ASTLiteral>(temp_column[0]), type_name);
+    Field literal;
+    /// If data type has custom serialization, we should use CAST from String,
+    /// because CAST from field may not work correctly (for example for type IPv6).
```

Comment:
The problem with IPv6 is that the underlying type is FixedString(16) and it's field type is String, so when we try to CAST field to IPv6 we try to do it through string parsing and get error, because this string is not an IPv6 string, it's 16 bytes representation of IPv6